decidim-core/lib/decidim/core/test/shared_examples/map_examples.rb

```diff
@@ -57,10 +57,16 @@ def snippets
 
 shared_context "with frontend map elements" do
   let(:html_head) { "" }
-  let(:html_document) do
+  let(:builder_snippets) do
     builder = subject
+    <<~HTML.strip
+      #{builder.append_assets}
+    HTML
+  end
+  let(:html_document) do
     document_inner = html_body
     head_extra = html_head
+    snippets = builder_snippets
```

Comment:
I think you can remove the snippets variable completely here because it's no longer returning anything as far as I understand. Just call `builder.append_assets` and you should be good to go.

And actually, I don't think you need to call it at all because you are already calling it when generating the map itself.

The `html_body` is generated above before calling the `x_pack_tag` helpers in the view below, so it should be already called by then.

Reply:
Actually we need this 

> 

I am trying this , but i have a feeling that it would complain about rendering order ;)